File: app/src/main/java/lheureux/votodroid/activities/QuestionAdapter.java
package lheureux.votodroid.activities;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import lheureux.votodroid.R;
import lheureux.votodroid.interfaces.onClickInterface;
import lheureux.votodroid.models.Question;

public class QuestionAdapter extends RecyclerView.Adapter<QuestionAdapter.MyViewHolder> {
    public List<Question> list;
    Context context;
    lheureux.votodroid.interfaces.onClickInterface onClickInterface;

    // Provide a reference to the views for each data item
    // Complex data items may need more than one view per item, and
    // you provide access to all the views for a data item in a view holder
    public static class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView tvQuestion;
        public LinearLayout questionContainer;
        public ImageButton btnResult;

        public MyViewHolder(LinearLayout v) {
            super(v);
            tvQuestion = v.findViewById(R.id.tvQuestion);
            questionContainer = v;
            btnResult = v.findViewById(R.id.btnResultButton);
        }
    }

    // Provide a suitable constructor (depends on the kind of dataset)
    public QuestionAdapter(Context context, List<Question> list, onClickInterface onClickInterface) {
        this.context = context;
        this.list = list;
        this.onClickInterface = onClickInterface;
    }

    // Create new views (invoked by the layout manager)
    @Override
    public QuestionAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent,
                                                         int viewType) {
        // create a new view
        LinearLayout v = (LinearLayout) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.question_item, parent, false);
        MyViewHolder vh = new MyViewHolder(v);
        Log.i("DEBOGAGE", "appel a onCreateViewHolder");
        return vh;
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        Question questionCourante = list.get(position);
        holder.tvQuestion.setText(questionCourante.text);
        holder.questionContainer.setOnClickListener(view -> {
            onClickInterface.setClick(0, questionCourante.questionId);
        });
        holder.btnResult.setOnClickListener(view -> {
            onClickInterface.setClick(1, questionCourante.questionId);
        });
        Log.i("DEBOGAGE", "appel a onBindViewHolder " + position);
    }

    // renvoie la taille de la liste
    @Override
    public int getItemCount() {
        return list.size();
    }

}

File: app/src/main/java/com/theodorelheureux/votodroid_android/activities/QuestionAdapter.kt
package com.theodorelheureux.votodroid_android.activities

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.theodorelheureux.votodroid_android.GetPaginatedQuery
import com.theodorelheureux.votodroid_android.R
import com.theodorelheureux.votodroid_android.activities.QuestionAdapter.MyViewHolder
import com.theodorelheureux.votodroid_android.interfaces.onClickInterface

class QuestionAdapter(
    var list: MutableList<GetPaginatedQuery.Question>,
    var onClickInterface: onClickInterface
) : RecyclerView.Adapter<MyViewHolder>() {
    class MyViewHolder(v: LinearLayout) : RecyclerView.ViewHolder(v) {
        var tvQuestion: TextView
        var questionContainer: LinearLayout
        var btnResult: ImageButton

        init {
            tvQuestion = v.findViewById(R.id.tvQuestion)
            questionContainer = v
            btnResult = v.findViewById(R.id.btnResultButton)
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): MyViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.question_item, parent, false) as LinearLayout
        val vh = MyViewHolder(v)
        return vh
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val question = list[position]
        holder.tvQuestion.text = question.text
        holder.questionContainer.setOnClickListener { view: View? ->
            onClickInterface.setClick(
                0,
                question.id.toString(),
                question.text
            )
        }
        holder.btnResult.setOnClickListener { view: View? ->
            onClickInterface.setClick(
                1,
                question.id.toString(),
                question.text
            )
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }
}